sys/shell/commands/sc_vfs.c

```diff
@@ -135,6 +138,21 @@ static int _df_handler(int argc, char **argv)
     return 0;
 }
 
+#if IS_USED(VFS_DEFAULT)
```

Comment:
Should be VFS_AUTO_MOUNT (it'd work just as well no matter whether these are defaults or set up by the application in a custom setup)

(also above in the corresponding help printf)

Reply:
How about we use `XFA_LEN` here instead?

Reply:
Does that work? I figure that if VFS_AUTO_MOUNT is not present, the relevant symbols are not even defined.

If it works, it's probably good enough (given that shell is not something present on the tiniest systems anyway). Note that MOUNTPOINTS_NUMOF gets its const value only really late, when the linker has assigned addresses and the symbols are substituted into the code -- I don't think that on NUMOF == 0, the code behind the check would even be recognized as dead code.

Reply:
Oh work in the sense of runtime it sure does; I was more wondering whether it still builds when VFS_AUTO_MOUNT is not an active module.

Reply:
This is `examples/filesystem` where `vfs_auto_mount` is not used

Reply:
Weird, looking into it to fix my understanding.

Reply:
Ah, found it: The XFA_INIT (and even its user auto_init_vfs) is in immaterial of whether the automounter is set, so whenever VFS is there, that array is too.

Conversely, this probably works even when VFS_DEFAULT is on but VFS_AUTO_MOUNT is off, which is not bad, just unexpected (and not even contradicting the documentation) -- because the macro named `VFS_AUTO_MOUNT` and the module `VFS_AUTO_MOUNT` do slightly different things.